test/odbc/psqlodbc/psqlodbc_tests_common.h

```diff
@@ -0,0 +1,337 @@
+#ifndef PSQLODBC_TESTS_COMMON_H
+#define PSQLODBC_TESTS_COMMON_H
+
+#include <cmath>
+#include <gtest/gtest.h>
+#include <sqlext.h>
+#include <string>
+#include <time.h>
+#include <vector>
+
+#include "../src/drivers.h"
+#include "../src/odbc_handler.h"
+#include "../src/query_generator.h"
+
+using std::vector;
+using std::string;
+using std::pair;
+
+const int BUFFER_LENGTH = 8192;
+const int BUFFER_SIZE = 256;
+const int CHARSIZE = 255;
+const int INT_BYTES_EXPECTED = 4;
+
+string InitializeInsertString(const vector<string>& insertedValues, bool isNumericInsert, int pkStartingValue = 0);
+
+string createTableConstraint(const string &constraintType, const vector<string> &constrainCols);
+
+void createTable(ServerType serverType, const string &table_name, const vector<pair <string, string>> &columns, const string &constraints = "");
+
+void createView(ServerType serverType, const string &viewName, const string &viewQuery);
+
+void dropObject(ServerType serverType, const string &objectType, const string &objectName);
+
+void insertValuesInTable(ServerType serverType, const string &table_name, const vector<string> &insertedValues, 
+  bool isNumericInsert, int pkStartingValue = 0);
+
+void insertValuesInTable(ServerType serverType, const string &tableName, const string &insertString, int numRows);
+
+void verifyValuesInObject(ServerType serverType, const string &objectName, const string &orderByColumnName, 
+  const vector<string> &insertedValues, const vector<string> &expectedInsertedValues, int pkStartingValue = 0);
+
+template <typename T>
+void verifyValuesInObject(ServerType serverType, string objectName, string orderByColumnName, int type, T data, 
+  int bufferLen, vector<string> insertedValues, vector<T> expectedInsertedValues, vector<long> expectedLen, int pkStartingValue = 0);
+
+void testCommonColumnAttributes(ServerType serverType, const string &tableName, int numCols, const string &orderByColumnName, 
+  const vector<int> &lengthExpected, const vector<int> &precisionExpected, const vector<int> &scaleExpected, const vector<string> &nameExpected);
+
+void testCommonCharColumnAttributes(ServerType serverType, const string &tableName, int numCols, const string &orderByColumnName, 
+  const vector<int> &lengthExpected, const vector<int> &precisionExpected, const vector<int> &scaleExpected, const vector<string> &nameExpected, 
+  const vector<int> &caseSensitivityExpected, const vector<string> &prefixExpected, const vector<string> &suffixExpected);
+
+void testTableCreationFailure(ServerType serverType, const string &tableName, const vector<vector<pair<string, string>>> &invalidColumns);
+
+void testInsertionSuccess(ServerType serverType, const string &tableName, const string &orderByColumnName, 
```

Comment:
Why do we need both testInsertionSuccess and the templated testInsertionSuccess? 
Similarly, testUpdateSuccess, testUpdateFail

And a few others testComparisonFunctions.

Can they be made one? Probably the templated one?

Reply:
I have tried to pass both string & other numeric types to the same template function, but unfortunately it looks like mixing numerics & strings in template functions don't play well:
```
error: no matching function for call to ‘testUpdateSuccess(...
```
So I opted to overload the template function with one that accepts string types. https://stackoverflow.com/questions/16184885/function-template-for-string-and-int-in-c